src/main.vala

```diff
@@ -346,7 +300,7 @@ class Vls.Server : Object {
         update_context_client = client;
         g_sources += Timeout.add (check_update_context_period_ms, () => {
             check_update_context ();
-            return !this.shutting_down;
+            return true;
```

Comment:
probably `return !cancellable.is_cancelled ();` would be better

Reply:
Yes, I agree.